engine/src/exec-extension.cpp

```diff
@@ -229,64 +229,45 @@ void MCEngineLoadExtensionFromData(MCExecContext& ctxt, MCDataRef p_extension_da
 }
 
 // This is the callback given to libscript so that it can resolve the absolute
-// path of native code libraries used by foreign handlers in the module. At
-// the moment we use the resources path of the module, however it will need to be
-// changed to a separate location at some point with explicit declaration so that
-// iOS linkage and Android placement issues can be resolved.
-//
-// Currently it expects:
-//   <resources>
-//     code/
-//       mac/<name>.dylib
-//       linux-x86/<name>.so
-//       linux-x86_64/<name>.so
-//       win-x86/<name>.dll
-//
+// path of native code libraries used by foreign handlers in the module.
+
 static bool MCEngineLoadLibrary(MCScriptModuleRef p_module, MCStringRef p_name, MCSLibraryRef& r_library)
 {
-    // If the module has no resource path, then it has no code.
-    MCAutoStringRef t_resource_path;
-    if (!MCEngineLookupResourcePathForModule(p_module, Out(t_resource_path)))
-        return false;
-    
     MCSLibraryRef t_library = nullptr;
-    if (t_resource_path.IsSet() &&
-        !MCStringIsEmpty(*t_resource_path))
+    
+    // extension libraries should be mapped by the IDE or deploy params
+    MCAutoStringRef t_mapped_path;
+    if (MCdispatcher->fetchlibrarymapping(p_name, &t_mapped_path))
```

Comment:
I presume the `t_mapped_path` is unused because you are just checking if there exists a library mapping here? If so could you add a comment to that effect (or an `MCDispatch::haslibrarymapping`) so we know it's intended

Reply:
good idea